src/data/dem_data.js

```diff
@@ -79,8 +79,12 @@ export default class DEMData {
         this._tree = new DemMinMaxQuadTree(this);
     }
 
-    get(x: number, y: number) {
+    get(x: number, y: number, clampToEdge: boolean = false) {
         const pixels = new Uint8Array(this.data.buffer);
+        if (clampToEdge) {
```

Comment:
When does it happen? Buildings that cross tile border have height encoded in centroid and there's no need to sample.

Reply:
For slope calculation, samples could be outside.
Lower clamp bound might need to be -1 (padded [0..511] range mapped to [-1..512]).

Reply:
Yes it happens for the slop calculation, as well as when elevation is sampled for a given vertex in `getTerrainHeightOffset` and when the vertex is outside of  tile boundaroes.

I updated the clamp ranges to account for padding.